packages/compiler/core/checker.ts

```diff
@@ -467,6 +467,21 @@ export function createChecker(program: Program): Checker {
     for (const decNode of augmentDecorators) {
       const ref = resolveTypeReferenceSym(decNode.targetType, undefined);
       if (ref) {
+        // disallow augmenting instantiated templates
+        if (ref.declarations.at(0)?.kind === SyntaxKind.ModelStatement) {
```

Comment:
There is technically quite a few other things that could be template that we should also check for

Reply:
Got rid of the logic that requires a model.